tasks/confidential_computing_tasks/homomorphic_encryption_tasks/homomorphic_security_algorithm.py

```
@@ -62,22 +62,24 @@ def multiply_messages(self, c1: T, c2: T) -> T:
     def scalar_and_message_multiplication(self, c: T, scalar: int) -> T:
         pass
 
-    def _extract_random_prime_p_and_q(self, key_file: str, num_of_key_parts: int = KeyConsts.DEFAULT_KEY_PARTS) -> \
+    def _extract_random_prime_p_and_q(self, key_file: str, should_generate: bool, num_of_key_parts: int = KeyConsts.DEFAULT_KEY_PARTS) -> \
```

You should either change the number typing to be optional, or raise n exception when the key file is not found

---

done, added exception

---

Do the callers should handle this exception?
If so, consider creating a custom exception

---

I don't think so... just let it fail the program, no?